feat(play): implement interactive chessboard and game modes

Cedarich opened the pull request:
### 🎯 Overview
This PR implements the main Play page featuring an interactive chessboard and game mode selection interface. The implementation focuses on creating a clean, modern UI while maintaining high functionality and user experience.

### 🛠️ Key Changes
- Created reusable ChessboardComponent with:
  
  - FEN string support for board positions
  - Basic move validation using chess.js
  - Responsive design with proper scaling
  - Touch and drag-n-drop support
- Implemented game mode selection with:
  
  - Modern gradient buttons with hover effects
  - Clear visual hierarchy
  - Intuitive icons and labels
  - Smooth animations and transitions
- Enhanced sidebar navigation:
  
  - Auto-collapse functionality
  - Smooth hover-based expansion
  - Improved visual feedback
  - Better mobile responsiveness
### 🎨 UI/UX Improvements
- Used a modern color scheme with teal and blue gradients
- Added subtle animations for better user feedback
- Implemented responsive layout for various screen sizes
- Enhanced accessibility with proper contrast and focus states
### 🧪 Testing
- Verified chessboard move handling
- Tested responsive behavior across different screen sizes
- Validated game mode selection functionality
- Confirmed sidebar navigation behavior
### 📝 Notes
- Game logic and turn validation will be implemented in future PRs
- AI agent integration points are prepared but not yet connected
- Mobile view optimizations may need further refinement
### 👀 Screenshots
![image](https://github.com/user-attachments/assets/c9ec4ec1-f89e-4c7c-a7d3-e5a46d5547c3)

### 🔄 Related Issues
Closes #[Issue number]

### 📋 Checklist
- Implemented basic chessboard functionality
- Added game mode selection UI
- Enhanced sidebar navigation
- Tested on multiple devices
- Added proper documentation
- Followed project coding standards
- Optimized for performance
- Ensured accessibility standards
Please review and provide feedback on the implementation approach and UI design choices.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced an interactive chessboard with drag-and-drop and click-based piece movement, responsive sizing, and keyboard accessibility.
  - Added visually engaging game mode selection buttons with animated effects and Lottie chess animation.
  - Implemented a dynamic chess game interface with real-time move handling and game mode switching.
  - Added a Lottie chess-themed animation asset for enhanced visual experience.

- **Improvements**
  - Enhanced the sidebar with hover-driven expansion, smoother transitions, unified navigation styling, and improved mobile layout.
  - Updated sidebar elements for consistent styling and animation across desktop and mobile.
  - Optimized logo rendering for mobile with improved image handling.

- **Chores**
  - Added new dependencies for chess logic, chessboard rendering, Lottie animations, and SVG/JSON asset support.
  - Updated TypeScript configuration and added module declarations for SVG and JSON imports.
  - Refactored Next.js configuration to enable strict mode and image unoptimization.

- **Style**
  - Reformatted TypeScript configuration for consistent code style.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jedstroke commented on frontend/components/chess/ChessboardComponent.tsx:
```diff
@@ -0,0 +1,328 @@
+    if (!piece) return null;
+
+    // Map piece codes to image paths
+    const pieceImages: Record<string, string> = {
```
Where are the images we prefer, chess.com type of pieces, not a Unicode fallback, which isn't visually appealing. That's what's blocking me from merging?

jedstroke replied:
![image](https://github.com/user-attachments/assets/526d6cc1-e0e3-4dbf-a7bc-56682fddcd9b)
Like this, see the pieces?

Cedarich replied:
Thanks for the feedback! The image assets (Chess.com-style pieces) will be sorted shortly

jedstroke reviewed (commented):
@Cedarich attend to these reviews.

Cedarich commented:
# Chess Piece Implementation Update
## Changes Made
- Replaced Unicode characters with SVG graphics
- Implemented color customization (#005dad for white pieces)
- Added proper scaling and transitions
- Integrated with Next.js Image optimization
## Current Status
The implementation uses free SVG chess pieces with custom styling. While not as polished as premium assets (like chess.com), they provide a solid foundation with:

- Clean vector graphics
- Color customization support
- Smooth animations
- Responsive scaling
## Note on Premium Assets
Professional chess piece designs (like those on chess.com) require:

- Paid licenses
- Subscriptions
- Custom commissions
For now, our SVG implementation offers the best free alternative while maintaining flexibility for future upgrades.

jedstroke commented:
I can't see your implemented changes (as touching the pieces), and also it should be on the start page. Instead of this `/play` it should be this `/`, and why is the sidebar overlaying the UI?

![image](https://github.com/user-attachments/assets/172f0842-010f-4501-be20-96011edb1876)
